src/services/expenses.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+
+let expenses = [];
+
+const initiate = () => {
+  expenses = [];
+};
+
+const getAll = ({
+  userId,
+  categories,
+  from,
+  to,
+}) => {
+  let filteredExpenses = [...expenses];
+
+  if (userId) {
+    filteredExpenses = filteredExpenses
+      .filter(e => e.userId === +userId);
+  }
+
+  if (categories) {
+    filteredExpenses = filteredExpenses
+      .filter(e => categories.includes(e.category));
+  }
+
+  if (from) {
+    filteredExpenses = filteredExpenses
+      .filter(e => e.spentAt > from);
```

Comment:
if few conditions are true you use filter few times , let's follow DRY & use filter just once

Reply:
Thanks, I made filter refactoring :)